src-tauri/src/filesystem/mod.rs

```diff
@@ -11,6 +15,25 @@ pub const fn bytes_to_gb(bytes: u64) -> u16 {
     (bytes / (1e+9 as u64)) as u16
 }
 
+/// Opens a file at the given path. Returns a string if there was an error.
+// NOTE(conaticus): I tried handling the errors nicely here but Tauri was mega cringe and wouldn't let me nest results in async functions, so used string error messages instead.
+#[tauri::command]
+pub async fn open_file(path: String) -> Result<String, ()> {
+    Ok(tokio::task::spawn_blocking(move || {
```

Comment:
You probably don't need to spawn an async task in an already Async command.

Also, if this can return a failure, you should use the `Err` result.

This means invocation on the client side will likely need a `.catch()` block.